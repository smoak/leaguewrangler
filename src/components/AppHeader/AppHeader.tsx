import AppBar from '@material-ui/core/AppBar';
import IconButton from '@material-ui/core/IconButton';
import { createStyles, withStyles, WithStyles } from '@material-ui/core/styles';
import Toolbar from '@material-ui/core/Toolbar';
import MenuIcon from '@material-ui/icons/Menu';
import React, { SFC } from 'react';
import { Typography } from '@material-ui/core';
import UserAvatar from '../UserAvatar';

const styles = createStyles({
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
  grow: {
    flexGrow: 1,
  },
});

export const AppHeader: SFC<WithStyles<typeof styles>> = ({ classes }) => {
  return (
    <AppBar position="static">
      <Toolbar>
        <IconButton className={classes.menuButton} color="inherit" aria-label="Menu">
          <MenuIcon />
        </IconButton>
        <Typography variant="h6" color="inherit" className={classes.grow}>
          {process.env.REACT_APP_WEBSITE_NAME}
        </Typography>
        <UserAvatar />
      </Toolbar>
    </AppBar>
  );
};

export default withStyles(styles)(AppHeader);
