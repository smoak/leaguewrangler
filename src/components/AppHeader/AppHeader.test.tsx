import { shallow, ShallowWrapper } from 'enzyme';
import { AppHeader } from './AppHeader';
import React from 'react';

describe('AppHeader', () => {
  let component: ShallowWrapper;

  beforeEach(() => {
    component = shallow(<AppHeader classes={{ menuButton: 'styles' }} />);
  });

  it('renders', () => {
    expect(component).toMatchSnapshot();
  });
});
