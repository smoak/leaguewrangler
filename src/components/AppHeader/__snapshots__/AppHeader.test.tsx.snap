// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader renders 1`] = `
<WithStyles(AppBar)
  position="static"
>
  <WithStyles(Toolbar)>
    <WithStyles(IconButton)
      aria-label="Menu"
      className="styles"
      color="inherit"
    >
      <pure(MenuIcon) />
    </WithStyles(IconButton)>
    <WithStyles(Typography)
      color="inherit"
      variant="h6"
    >
      Team Cowgirl
    </WithStyles(Typography)>
    <Apollo(WithStyles(UserAvatar)) />
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;
